### PR TITLE
Optimize nemotron VL image/video preprocessing

### PR DESCRIPTION
**Pull request by milesial**
## Purpose

Compile and reorganize image/video preprocessing for nemotron nano VL, reducing the amount of CPU time and memory needed.

- Fused resize+normalize+cast under @torch.compile — CPU kernel for permute → bicubic → /255 → (x-mean)/std → dtype.
- dtype conversion integrated in the fusion to avoid a later separate autocast
- contiguous fused to avoid a later separate H2H copy
- Skip torch.cat on the single-image / single-video path to avoid a redundant copy
- Batched tokenizer call for video frame separators

```
  1 video of 512x512x512, H100
Before:     apply_hf_processor_ms 898.57 898.63 4.58 905.18
After:       apply_hf_processor_ms 254.21 254.56 3.35 260.79
```

### CODE REVIEW

**Review by netanel-haber — APPROVED** [review]
LGTM. I ran evals and VoxPopuli (audio+text), InfoVQA_VAL (image+text) and DailyOmni (video+audio+text) are on par before and after.
